### ISSUE TITLE
Maven: "Theme name and theme class can not both be specified" with Material theme and themeFolder

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#15746]
### Description of the bug

Vaadin fails to build the production bundle for the annotation `@Theme(value = Material.class, themeFolder = "my-theme")`, with a strange error message:
```
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:14.9.5:build-frontend (default) on project project-base: Execution default of goal com.vaadin:vaadin-maven-plugin:14.9.5:build-frontend failed: Theme name and theme class can not both be specified. Theme name uses Lumo and can not be used in combination with custom theme class that doesn't extend Lumo. -> [Help 1]
```

If the themeFolder attribute isn't supported in connection with the Material theme, the error message should mention themeFolder instead of theme name.

### Expected behavior

Vaadin should either support Material custom theme according to the https://vaadin.com/docs/v14/flow/styling/custom-theme manual, or Maven should fail with a clearer error message.

### Minimal reproducible example

1. clone skeleton-starter-flow for Vaadin 14
2. add `@Theme(value = Material.class, themeFolder = "my-theme")` to the `MainView` class
3. `mvn -C clean package -Pproduction`

### Versions

- Vaadin / Flow version: 14.9.5
- Java version: 11

**Comment by knoobie** [comment]
Looking at the linked docs they state as limitation:

> Using the built-in Material theme is not currently supported. Custom themes are always loaded on top of the Lumo theme.

**Comment by mvysny** [comment]
Yup, you're right, the documentation clears up everything. Still, the Maven plugin error message could be improved to mention the custom themes and Lumo. Essentially, the same line as from the documentation would be great.
